server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1426,6 +1430,14 @@ private SearchSourceBuilder parseXContent(
                     knnBuilders = List.of(KnnSearchBuilder.fromXContent(parser));
                     searchUsage.trackSectionUsage(KNN_FIELD.getPreferredName());
                 } else if (RANK_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    if (parser.getRestApiVersion().matches(RestApiVersion.onOrAfter(RestApiVersion.V_8))) {
+                        deprecationLogger.warn(
+                            DeprecationCategory.API,
+                            "sub_searches_api",
+                            "Deprecated field [rank] used, this field is unused and will be removed entirely in a future release. "
+                                + "Please consider using the appropriate [retriever] to rank your results."
+                        );
+                    }
```

Comment:
I don't think we need to add any logger as the deprecation on the ParseField should do it. Does that not work?

Reply:
Oh yeah, you're right.. Hadn't pulled all changes to my local branch before testing 🤦  Removed :) 

If we add a reference to `retriever` at the `ParseField` deprecation, the message is now:
```
Deprecated field [sub_searches] used, replaced by [retriever]
```
whilst if we do not provide a replacement  it is
```
Deprecated field [sub_searches] used, this field is unused and will be removed entirely
```

Reply:
@pmpailis you can also just provide your own full deprecation message :)

Reply:
@pmpailis with replacement to retriever makes sense to me

Reply:
Updated in fa699e125101cf6b34adc3b10b66a5660f9d5109. (was also looking for custom message to be a bit more expressive)